experiments/modern/src/maki-interpreter/reference/maki_compiler/v1.1.1.b3 (Winamp 3.0 build 488d)/simpleFunctions.m

```diff
@@ -1,28 +1,101 @@
-#include "lib/std.mi"
-
-Function addInts(int a, int b);
-
-System.onScriptLoaded()
-{
-    Int two = 2;
-	Int sum = addInts(two, two);
-    if (sum == 4) {
-        messageBox("simple custom function", "Success", 0, "");
-    } else {
-        messageBox("simple custom function", "Fail", 1, "");
-    }
-
-    Double val = 2.2;
-	sum = addInts(two, val);
-    if (sum == 4) {
-        messageBox("simple custom function with implicit cast", "Success", 0, "");
-    } else {
-        messageBox("simple custom function with implicit cast", "Fail", 1, "");
-    }
-}
-
-addInts (int a, int b)
-{
-    Int sum = a + b;
-    return sum;
-}
+#include "lib/std.mi"
+
+Global int num = 0;
+
+Function addInts(int a, int b);
+Function fib(int n);
+
+System.onScriptLoaded()
+{
+    Int two = 2;
+	Int sum = addInts(two, two);
+    if (sum == 4) {
+        messageBox("simple custom function", "Success", 0, "");
+    } else {
+        messageBox("simple custom function", "Fail", 1, "");
+    }
+
+    Double val = 2.2;
+	sum = addInts(two, val);
+    if (sum == 4) {
+        messageBox("simple custom function with implicit cast", "Success", 0, "");
+    } else {
+        messageBox("simple custom function with implicit cast", "Fail", 1, "");
+    }
+
+    if (fib(2) == 0) {
+        messageBox("fib(2) == 0 (should be 1)", "Success", 0, "");

```

Comment:
What do you think about adding a comment here about what we think is going on?

Reply:
Definitely!